### PR TITLE
Move driver-toolkit e2e test to release-4.x configs

### PR DESCRIPTION
**Pull request by dagrayvid**
Signed-off-by: David Gray <dagray@redhat.com>

/cc @kpouget

### CODE REVIEW

**Comment by kpouget** [comment]
thanks @dagrayvid 
/lgtm

**Review by pacevedom — APPROVED** [review]

**Comment by smarterclayton** [comment]
Why is the `-src` being published?  Generally `-src` images stay within the project environment since others do not use them.

**Comment by dagrayvid** [comment]
> Why is the -src being published? Generally -src images stay within the project environment since others do not use them.

There will be a follow-up PR to add the job to `ci-operator/config/openshift/release/openshift-release-master__nightly-4.*.yaml`, using the `driver-toolkit-src` images. Our intention is to make the driver-toolkit e2e test a release-informing job.

**Comment by dagrayvid** [comment]
> Why is the -src being published? Generally -src images stay within the project environment since others do not use them.

@smarterclayton There will be a follow-up PR to add the job to ci-operator/config/openshift/release/openshift-release-master__nightly-4.*.yaml, using the driver-toolkit-src images. Our intention is to make the driver-toolkit e2e test a release-informing job.

**Comment by smarterclayton** [comment]
If the goal is to consume the contents of this image to perform tests, the image should be called `-tests`.  Exporting it as `-src` doesn't make it clear what the goal is, whereas exporting `driver-toolkit-tests` is perfectly clear.  You should be able to simply map `src` to `tests` in your build and then export it (you may possibly have to add a no-op addition), but if you hit a snag we'll work through it.  That makes it clear that the goal of this image is to perform tests, and in the future if you design new types of tests you can add them to this image (whereas adding them to src would be incorrect).

**Comment by dagrayvid** [comment]
> If the goal is to consume the contents of this image to perform tests, the image should be called `-tests`. Exporting it as `-src` doesn't make it clear what the goal is, whereas exporting `driver-toolkit-tests` is perfectly clear. 

Good point. Thanks

> You should be able to simply map `src` to `tests` in your build and then export it (you may possibly have to add a no-op addition), but if you hit a snag we'll work through it. That makes it clear that the goal of this image is to perform tests, and in the future if you design new types of tests you can add them to this image (whereas adding them to src would be incorrect).

Is it sufficient to simply export the src image as `driver-toolkit-test`, since the src image contains all the resources needed to run the tests?
```
promotion:
  additional_images:
    driver-toolkit-test: src
 ```
 
 This seems simpler than mapping `src` to `test` in the build

**Comment by smarterclayton** [comment]
Yes that is acceptable

**Comment by dagrayvid** [comment]
/retest

Unclear to me why the release-4.9/e2e-upgrade test is failing... Still hoping it is a flake

**Comment by dagrayvid** [comment]
/test pj-rehearse

From discussing on slack, it seems like a flake / issue on build03

**Comment by dagrayvid** [comment]
I don't think we need these tests to merge. Definitely seem to be flakes.

/test pj-rehearse

**Comment by kpouget** [comment]
thanks @dagrayvid 
/lgtm

File: core-services/image-mirroring/openshift/mapping_origin_4_9 [hunk 1]
```diff
@@ -98,6 +98,7 @@ registry.ci.openshift.org/origin/4.9:descheduler quay.io/openshift/origin-desche
 registry.ci.openshift.org/origin/4.9:docker-builder quay.io/openshift/origin-docker-builder:4.9 quay.io/openshift/origin-docker-builder:4.9.0
 registry.ci.openshift.org/origin/4.9:docker-registry quay.io/openshift/origin-docker-registry:4.9 quay.io/openshift/origin-docker-registry:4.9.0
 registry.ci.openshift.org/origin/4.9:driver-toolkit quay.io/openshift/origin-driver-toolkit:4.9 quay.io/openshift/origin-driver-toolkit:4.9.0
+registry.ci.openshift.org/origin/4.9:driver-toolkit-test quay.io/openshift/origin-driver-toolkit-test:4.9 quay.io/openshift/origin-driver-toolkit-test:4.9.0
```
Comment (smarterclayton) [inline core-services/image-mirroring/openshift/mapping_origin_4_9:101]: driver-toolkit-tests

(we use plural “tests”)

Comment (dagrayvid) [reply core-services/image-mirroring/openshift/mapping_origin_4_9]: Done, thanks.

**Pull request merged**